R/uvhydrograph-fnc.R

```diff
@@ -1,133 +1,131 @@
 
 #'@export
 uvhydrographPlot <- function(data){
-  uv_pts <- getUvHydro(data, "primarySeries" )
   layout_uvhydro()
   
+  #legend vector, needs to be dynamically built up with plots
+  primary_legend <- getNewLegendFrame()
+  addToPrimaryLegend <- function(newText, newSymbol, newColor, newLine) { 
+    primary_legend <<- rbind(primary_legend, data.frame(text = newText, symbol = newSymbol, color = newColor, line = newLine, stringsAsFactors = FALSE))
```

Comment:
No, not too bad. 

Here's far more information than needed, if interested:

http://adv-r.had.co.nz/Functional-programming.html#closures
http://adv-r.had.co.nz/Environments.html
and finally:
http://adv-r.had.co.nz/OO-essentials.html#rc